plugins/tours/tests/phpunit/integration/registerCustomConfigs.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Tests for register_custom_configs().
+ *
+ * @since       1.0.0
+ * @author      Robert Gadon <rgadon107>
+ * @package     spiralWebDb\CornerstoneTours\Tests\Integration
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\CornerstoneTours\Tests\Integration;
+
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+use function spiralWebDb\CornerstoneTours\register_custom_configs;
+
+/**
+ * @covers ::\spiralWebDb\CornerstoneTours\register_custom_configs
+ *
+ * @group   tours
+ */
+class Tests_RegisterCustomConfigs extends Test_Case {
+
+	/**
+	 * Test register_custom_configs() is registered to filter 'add_custom_post_type_runtime_config' when event fires.
+	 */
+	public function test_callback_is_registered_to_filter_hook_when_event_fires() {
+		$this->assertEquals( 7, has_filter( 'add_custom_post_type_runtime_config', 'spiralWebDb\CornerstoneTours\register_custom_configs' ) );
+	}
+
+	/**
+	 * @dataProvider addTestData
+	 * @param array $configurations
+	 * @param array $tours
+	 */
+	public function test_should_add_runtime_post_type_config_to_existing_configs( array $configurations, array $tours) {
+
+		$this->assertArrayHasKey( 'tours', register_custom_configs( (array) $configurations ) );
+		$this->assertContains( $tours['post_type'], (array) $tours );
+		$this->assertContains( $tours['features'], (array) $tours );
+		$this->assertContains( $tours['args'], (array) $tours );
+	}
```

Comment:
Notice that the `register_custom_configs()` is registered to a `add_custom_post_type_runtime_config` filter event. It's a callback. That means you don't directly invoke it; rather, it's called automatically when that filter event fires.

What fires that filter event and when? 

It's fired in the [Central Hub's custom module](https://github.com/rgadon107/cornerstone/blob/registerCustomConfigs/mu-plugins/central-hub/src/custom/post-type.php#L39).  Notice it happens on `init`.

The thing to know about `'init'` is that it has already fired by the time this test runs. 

Hmm 🤔 then how do you trigger loading the custom configs for this test? Fire the filter event in the test like this:

```php
public function test_should_add_runtime_post_type_config_to_existing_configs( array $configurations, array $tours) {
	$configs = apply_filters( 'add_custom_post_type_runtime_config', [] );

	//
}
```

Then the tests

- You want to validate the tours config was successfully merged into the existing configurations array. 
- To do that, you check it has the tour as a key.
- Then compare the expected with the actual.

```php
public function test_should_add_runtime_post_type_config_to_existing_configs( array $existing, array $expected ) {
$configs = apply_filters( 'add_custom_post_type_runtime_config', $existing );

$this->assertArrayHasKey( 'tours', $configs );
$this->assertSame( $expected, $configs['tours'] );
```

What if there are existing configs? You check that by adding existing configs to the test data and then checking for those:

```php
public function test_should_add_runtime_post_type_config_to_existing_configs( array $existing, array $expected ) {
	$configs = apply_filters( 'add_custom_post_type_runtime_config', $existing );

	$this->assertArrayHasKey( 'tours', $configs );
	$this->assertSame( $expected, $configs['tours'] );

	foreach ( $existing as $key => $config ) {
		$this->assertArrayHasKey( $key, $configs );
		$this->assertSame( $config, $configs[ $key ] );
	}
}
```

Reply:
I updated the integration test in this commit https://github.com/rgadon107/cornerstone/pull/68/commits/3e580589b2414993b1793d2f49a8ad4a3cada1a5.

Reply:
@hellofromtonya There's a lot going on here. What I found initially confusing is the use of the data provider parameters `$existing` and `$expected` passed to the test method. 

I used `var_dump()` to inspect the value returned by each variable, and then compared them to the data sets in the data provider method. That helped to discern what was going on in the test method.

Reply:
The `apply_filters` for that event sends out the existing custom configs. In the source code, it's an empty array. But here, we are validating the behavior by passing it in from the tests data. That gives you the flexibility to further validate the behavior.

Did you notice the pattern between the data provider's test data array and the arguments passed to the test method?

```php
public function addTestData() {
	$tours = (array) require _get_plugin_directory() . '/config/post-type.php';

	return [

		// This is the 1st scenario's tests data.
		'testShouldIncludToursConfig' => [
			'existing_configs' => [], // 👈this is the 1st arg passed to the test method
			'tours'            => $tours, // 👈this is the 2nd arg passed to the test method
		],

		// This is the 2nd scenario's tests data.
		'testShouldMergeToursWithExistinConfigs' => [
			'existing_configs' => [  // 👈this is the 1st arg passed to the test method
				'ipseum' => [
					'post_type' => 'ipseum',
					'labels'    => [
						'singular_label' => 'Ipseum',
					],
					'features'  => [
						'base_post_type' => 'post',
					],
					'args'      => [],
				],
			],
			'tours'            => $tours, // 👈this is the 2nd arg passed to the test method
		],
	];
}
```

Let's break it down:

```php
// This is the 1st scenario's tests data.
'testShouldIncludToursConfig' => [
	'existing_configs' => [], // 👈Existing configs array that is passed with the filter.
	'tours'            => $tours, // 👈Expected tours configuration that should be contained within the configurations that are built and returned after the filter runs.
],
```

Does that make sense?

Reply:
@hellofromtonya 
>`Does that make sense?`

Yes, finally. I stared, and stared at the screen for hours yesterday and did not see the `require` to the `tours` post-type config array until just now. Guess it just takes a rested brain to make sense of things sometimes. I kept wondering where the var `$tours` was originating from. Now I see it.